flows/index.py

```diff
@@ -649,30 +676,55 @@ async def run_partial_updates_of_concepts_for_document_passages(
                     )
                     for concept in concepts
                 ]
-                concepts_counts.update(concepts_models)
-
-        # Write concepts counts to S3
-        try:
-            s3_uri = Path(document_importer[1])
-            # Get all parts after the prefix (e.g. "Q787/v4/CCLW.executive.1813.2418.json")
-            key_parts = "/".join(s3_uri.parts[3:])  # Skip s3:/bucket/labelled_passages/
-
-            # Create new path with concepts_counts_prefix
-            concepts_counts_uri = (
-                f"s3://{cache_bucket}/{concepts_counts_prefix}/{key_parts}"
-            )
 
-            serialised_concepts_counts = json.dumps(
-                {str(k): v for k, v in concepts_counts.items()}
-            )
+                match change:
+                    case Change.ADD_AND_UPDATE:
+                        concepts_counts.update(concepts_models)
+                    case Change.REMOVE:
+                        concepts_counts.subtract(concepts_models)
+
+                # Possibly remove the inference results from S3
```

Comment:
Yeah I'm not majorly worried about s3 removal at the moment, I can't seem to think of any worrying reason if they persist as we largely either rely on params passed into the flow or the config files for the model config 🤔

Reply:
The default run isn't discerning, as in, we don't pass params, and it just grabs everything [1]. So we'd have to change that behaviour, if we didn't delete things?

[1] https://github.com/climatepolicyradar/knowledge-graph/blob/feat%2Fdeindex/flows/index.py#L1111-L1113

Reply:
Resolving as I don't think this is doable without those changes! Happy to consider otherwise though 🙇🏻